### PR TITLE
Generate an initfile for every directory.

### PR DESCRIPTION
**Pull request by landrito**

### CODE REVIEW

File: protoc_docs/bin/py_docstring.py [hunk 1]
```diff
@@ -53,5 +54,20 @@ def main(input_file=sys.stdin, output_file=sys.stdout):
...
             ))
+    for fn in _init_files(comment_data.keys()):
+        answer.append(CodeGeneratorResponse.File(
+            name=fn,
+            content='',
+        ))
     cgr = CodeGeneratorResponse(file=answer)
     output_file.write(cgr.SerializeToString())
+
+def _init_files(fns=[]):
```
Comment (lukesneeringer) [inline protoc_docs/bin/py_docstring.py]: Do not use empty list as a default argument in Python. It creates a single instance which, if modified, would carry over from invocation to invocation.

Use empty tuple in this case.

File: tests/test_py_docstring.py [hunk 2]
```diff
@@ -58,4 +58,9 @@ def test_real_input_file(self):
 
         # Just ensure that the bytestream is the appropriate length.
         # This is a terrible test. :-/
-        assert len(output_file.getvalue()) == 25294
+        assert len(output_file.getvalue()) == 25318
```
Comment (lukesneeringer) [inline tests/test_py_docstring.py:61]: Mea culpa.

File: tests/test_py_docstring.py [hunk 3]
```diff
@@ -58,4 +58,9 @@ def test_real_input_file(self):
...
+
+    def test_init_files(self):
+        files = ['foo.proto', '/bar.proto', 'baz/qux/corge.proto']
+        expected = {'__init__.py', 'baz/qux/__init__.py'}
```
Comment (lukesneeringer) [inline tests/test_py_docstring.py:65]: Oh, this is odd. I know why it does this; we do not want to generate init files above the protos, but it is strange that in this case we generate the root and leaf, but not the branch.

Whatever. Leave as-is.

Comment (landrito) [reply tests/test_py_docstring.py:65]: Gotcha. 


> Leave as-is

👀 Nice pun 👍

File: protoc_docs/bin/py_docstring.py (hunk 1 above)
Comment (landrito) [reply protoc_docs/bin/py_docstring.py]: Oh wow. That kind of error looks terribly simple enough to be very common. Fixed.

Comment (lukesneeringer) [reply protoc_docs/bin/py_docstring.py]: Yeah, I am surprised that was not fixed in Python 3.

**Review by lukesneeringer — APPROVED** [review]

**Comment by mmellison** [comment]
BTW, this is not a good feature when generating Protobuf files into an existing source tree with custom __init__ files already in place, as it destroys the existing ones. :(

**Pull request merged**